Feature/view log out

CalebHasch opened the pull request:
## Describe your changes
Added a settings route with only a log out and delete profile button added to the page.

## Issue ticket number and link
ticket#: DROP-27
link: https://tripleten-apiary.atlassian.net/jira/software/projects/DROP/boards/5?selectedIssue=DROP-27

## Checklist before requesting a review

- [x] I have performed a self-review of my code
- [ ] If there are changes to components, I have added / updated automated tests
- [ ] I have made any necessary updates to Storybook to accompany these changes
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

Aivaxela reviewed (approved):
All styling and functionality looks good to me 👍

RPinkha commented on src/routes/settings.tsx:
```diff
@@ -0,0 +1,20 @@
+
+export const Settings: React.FC = () => {
+  return (
+<div className="flex flex-col bg-[#F5F5F5]">
```
You have the background color in the config file. 
bg-lightGray is better.

RPinkha commented on src/routes/settings.tsx:
I am thinking since this file is a component, you can make a folder called settings in the routes directory and put this file in there.

RPinkha reviewed (approved):
one minor issue I found.
The other one is that, the buttons have a padding by default which is fine, but the bottom padding of this component always seems to be 60px. Maybe add that in.
I would also remove the top padding so it will be easier to code in the distances later when this component is implemented. 

Minor fixes that aren't a big deal, so I am approving.

Looks good otherwise.

RPinkha reviewed (commented):
I just noticed that your buttons span the whole width of the screen.
I think their width needs to be restricted to prevent accidental clicks.

CalebHasch commented:
> I just noticed that your buttons span the whole width of the screen. I think their width needs to be restricted to prevent accidental clicks.

I fixed this within the settings file, but is this something that would be better if we changed it in the button component itself? Same question for the padding?

RPinkha commented:
`I fixed this within the settings file, but is this something that would be better if we changed it in the button component itself? Same question for the padding?`

@CalebHasch can you commit and push? 
Components usually have a classname prop passed to them for extra styling that one can use to style a component. That way we don't have to keep going back to fix things all the time.

RPinkha commented on src/routes/settings/settings.tsx:
```diff
@@ -0,0 +1,22 @@
+
+export const Settings: React.FC = () => {
+  return (
+    <div className="flex flex-col bg-lightGray items-center gap-8">
```
This div is unnecessary. You can remove it completely and the closing tag on line 20.

RPinkha commented on src/routes/settings/settings.tsx:
```diff
@@ -0,0 +1,22 @@
+export const Settings: React.FC = () => {
+  return (
+    <div className="flex flex-col bg-lightGray items-center gap-8">
+      <div className="flex flex-col items-center gap-8 my-[60px]">
```
Add bg-lightGray here. 
gap-8 is not needed if you restore button.tsx back to how it was.
my-[60px] adds a margin of 60 on top and on bottom, which isn't what we want. 
Use pb for padding bottom, the value would be 11 (since we restored button.tsx back to have py-4, you can subtract  16px from 60px which is 44px, which corresponds to 11 REM).

The whole line should look like this:
<div className="flex flex-col items-center pb-11 bg-lightGray">
And would look identical to how it was before, if my previous comments were implemented.

RPinkha commented on src/routes/settings/settings.tsx:
```diff
@@ -0,0 +1,22 @@
+      <div className="flex flex-col items-center gap-8 my-[60px]">
+        <Button
+          variant="text"
+          className="opacity-100 hover:opacity-60 text-base max-w-max"
```
add pt-0 to override the component top padding.

RPinkha reviewed (commented):
A few changes to implement and you should be good to go :)

RPinkha reviewed (approved):
LGTM
